File: design_patterns_study/230407_factory/go/abstractFactory/polygon/polygon.go
package polygon

import "fmt"

type Polygon interface {
	Get()
}

type BlueTriangle struct{}

func (t BlueTriangle) Get() {
	fmt.Println("Hello from a BlueTriangle")
}

type RedTriangle struct{}

func (t RedTriangle) Get() {
	fmt.Println("Hello from a RedTriangle")
}

type GreenTriangle struct{}

func (t GreenTriangle) Get() {
	fmt.Println("Hello from a GreenTriangle")
}

type BlueRectangle struct{}

func (r BlueRectangle) Get() {
	fmt.Println("Hello from a BlueRectangle")
}

type RedRectangle struct{}

func (r RedRectangle) Get() {
	fmt.Println("Hello from a RedRectangle")
}

type GreenRectangle struct{}

func (r GreenRectangle) Get() {
	fmt.Println("Hello from a GreenRectangle")
}

type BluePentagon struct{}

func (p BluePentagon) Get() {
	fmt.Println("Hello from a BluePentagon")
}

type RedPentagon struct{}

func (p RedPentagon) Get() {
	fmt.Println("Hello from a RedPentagon")
}

type GreenPentagon struct{}

func (p GreenPentagon) Get() {
	fmt.Println("Hello from a GreenPentagon")
}

File: design_patterns_study/230407_factory/rust/abstract_factory/project/src/polygon.rs

pub trait Polygon{
    fn get(&self);
}

pub struct BlueTriangle {}

impl Polygon for BlueTriangle {
    fn get(&self) {
        println!("Hello from BlueTriangle");
    }
}

pub struct RedTriangle {}

impl Polygon for RedTriangle {
    fn get(&self) {
        println!("Hello from RedTriangle");
    }
}

pub struct GreenTriangle {}

impl Polygon for GreenTriangle {
    fn get(&self) {
        println!("Hello from GreenTriangle");
    }
}

pub struct BlueRectangle {}

impl Polygon for BlueRectangle {
    fn get(&self) {
        println!("Hello from BlueRectangle");
    }
}

pub struct RedRectangle {}

impl Polygon for RedRectangle {
    fn get(&self) {
        println!("Hello from RedRectangle");
    }
}

pub struct GreenRectangle {}

impl Polygon for GreenRectangle {
    fn get(&self) {
        println!("Hello from GreenRectangle");
    }
}

pub struct BluePentagon {}

impl Polygon for BluePentagon {
    fn get(&self) {
        println!("Hello from BluePentagon");
    }
}

pub struct RedPentagon {}

impl Polygon for RedPentagon {
    fn get(&self) {
        println!("Hello from RedPentagon");
    }
}

pub struct GreenPentagon {}

impl Polygon for GreenPentagon {
    fn get(&self) {
        println!("Hello from GreenPentagon");
    }
}